### PR TITLE
KAFKA-12590: Remove deprecated kafka.security.auth.Authorizer, SimpleAclAuthorizer and related classes in 3.0

### PR DESCRIPTION
**Pull request by ijuma**
These were deprecated in Apache Kafka 2.4 (released in December 2019) to be replaced
by `org.apache.kafka.server.authorizer.Authorizer` and `AclAuthorizer`.

As part of KIP-500, we will implement a new `Authorizer` implementation that relies
on a topic (potentially a KRaft topic) instead of `ZooKeeper`, so we should take the chance
to remove related tech debt in 3.0.

Details on the issues affecting the old Authorizer interface can be found in the KIP:
https://cwiki.apache.org/confluence/display/KAFKA/KIP-504+-+Add+new+Java+Authorizer+Interface

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
@rajinisivaram @omkreddy Thoughts on this? Do you know if third-party implementations have migrated to the new interface?

**Comment by omkreddy** [comment]
looks like [Apache Ranger](https://github.com/apache/ranger/blob/master/plugin-kafka/src/main/java/org/apache/ranger/authorization/kafka/authorizer/RangerKafkaAuthorizer.java) is not migrated to the new interface. 

maybe we can remove and send a mail to the mailing list?

**Comment by ijuma** [comment]
@omkreddy Thanks. Maybe we can file a JIRA. Their code would probably be quite a bit simpler by not having to deal with the Scala conversions.

**Comment by ijuma** [comment]
Also, did Sentry migrate?

**Comment by omkreddy** [comment]
Sentry project has retired. so no issues i think.

**Comment by ijuma** [comment]
I filed a JIRA for Apache Ranger:
https://issues.apache.org/jira/browse/RANGER-3231

**Comment by ijuma** [comment]
@omkreddy I updated the PR description to elaborate a bit more on the motivation. I agree we should go ahead with this PR. Can you review it, please?

File: docs/upgrade.html [hunk 1]
```diff
@@ -27,6 +27,8 @@ <h5><a id="upgrade_300_notable" href="#upgrade_300_notable">Notable changes in 3
         or updating the application not to use internal classes.</li>
     <li>The Streams API removed all deprecated APIs that were deprecated in version 2.5.0 or earlier.
         For a complete list of removed APIs compare the detailed Kafka Streams upgrade notes.</li>
+    <li>The deprecated Scala <code>Authorizer</code><code>, <code>SimpleAclAuthorizer</code> and related classes have been removed. Please use the Java <code>Authorizer</code>
```
Comment (rondagostino) [inline docs/upgrade.html]: extra/unbalanced opening tag at end of `<code>Authorizer</code><code>`

File: tests/kafkatest/tests/core/security_rolling_upgrade_test.py [hunk 2]
```diff
@@ -93,8 +93,8 @@ def roll_in_sasl_mechanism(self, security_protocol, new_sasl_mechanism):
         self.kafka.interbroker_sasl_mechanism = new_sasl_mechanism
         self.bounce()
 
-        # Bounce again with ACLs for new mechanism. Use old SimpleAclAuthorizer here to ensure that is also tested.
-        self.set_authorizer_and_bounce(security_protocol, security_protocol, KafkaService.SIMPLE_AUTHORIZER)
+        # Bounce again with ACLs for new mechanism.
+        self.set_authorizer_and_bounce(security_protocol, security_protocol, KafkaService.ACL_AUTHORIZER)
```
Comment (rondagostino) [inline tests/kafkatest/tests/core/security_rolling_upgrade_test.py]: Can invoke `self.set_authorizer_and_bounce(security_protocol, security_protocol)` as is done elsewhere in this file and eliminate the ability to specify a different authorizer class name in the method, like this:

```
    def set_authorizer_and_bounce(self, client_protocol, broker_protocol):
        self.kafka.authorizer_class_name = KafkaService.ACL_AUTHORIZER
```

Comment (ijuma) [reply tests/kafkatest/tests/core/security_rolling_upgrade_test.py]: Thanks, updated.

**Review by omkreddy — APPROVED** [review]
@ijuma Thanks for the PR. LGTM.

**Pull request merged**